### DISCUSSION TITLE
New Install Pump Communication Questions

### DISCUSSION THREAD

**Discussion by notmydayjob** [tagyoureit/nodejs-poolController#779] [RS485] [answered]
I have just deployed NJspC to replace a homebrew python based system. I have gotten my head around most of the configuration but I am having an issue with serial communications with my Itelleflo VS pump. I can control the pump just fine, start, stop, adjust speed etc. However it is not displaying the speed or wattage on the dash panel. The odd thing is I am pretty sure it was displaying this info when testing, but now that is finally deployed it is not. I double checked my serial cable wiring and port settings.

I am using a ComfilePI (http://comfilewiki.co.kr/en/doku.php?id=comfilepi:index) which is and industrial RPi automation controller that includes an RS-285 among other things. 

>  "(*) COM1 (/dev/serial1) on the ComfilePi A & B series (CPi-A & CPi-B) utilizes the Raspberry Pi's mini UART. The mini UART has a 
> few limitations (e.g. no parity bit). For more information please see the [Raspberry Pi documentation]"

I have attached a set of logs, during the logging period I power cycled the pump from the control panel in an effort to ensure sufficient traffic between the controller and the pump.

Thanks for your help

[replay.zip](https://github.com/tagyoureit/nodejs-poolController/files/11675281/replay.zip)

**Comment by rstrouse** [comment]
In looking at your replay I was very confused it looked like the pump was just responding with whatever was sent to it but that is not the case.  The comms are simply echoing back from the port and not receiving any bytes.  This is why it is able to change the speed but no status is returned.  Not sure about the configuration of the board but if you have it set to echo somewhere this is your problem.

**Reply by notmydayjob** [reply, marked as answer]
TLDR 

OPERATOR ERROR

Full story.
I have two "identical" ComfilePi controllers, One was installed, and the second I purchased as a spare. I used the spare on my bench to configure and test NJSPC.  When this was complete I pulled the SD card from the unit on my bench and swapped it into the installed unit.

Upon closer inspection I did not, in fact have two identical units, The one that was installed was an older revision that had two RS232 ports, while the one of my bench had an RS232 and an RS485.  After swapping the units, communication between the pump and the controller is behaving as expected.

Thanks for your help. Sorry to waste your time.
